core/objectspace/define_finalizer_spec.rb

```diff
@@ -168,6 +168,31 @@ def finalizer(zelf)
     ruby_exe(code).lines.sort.should == ["finalized1\n", "finalized2\n"]
   end
 
+  it "defines same finalizer only once" do
+    code = <<~RUBY
+      obj = Object.new
+      p = proc { |id| print "ok" }
+      ObjectSpace.define_finalizer(obj, p.dup)
+      ObjectSpace.define_finalizer(obj, p.dup)
+    RUBY
+
+    ruby_exe(code).should == "ok"
+  end
+
+  it "returns the defined finalizer" do
+    obj = Object.new
+    p = proc { |id| }
+    p2 = p.dup
+
+    ret = ObjectSpace.define_finalizer(obj, p)
+    ret.should == [0, p]
+    ret[1].should.equal?(p)
+
+    ret = ObjectSpace.define_finalizer(obj, p2)
+    ret.should == [0, p]
+    ret[1].should.equal?(p)
```

Comment:
I wonder if these semantics are really intended, it seems weird to treat two different Proc instances as the same "finalizer". Although at least these Procs are `==`.
Maybe something to discuss at https://bugs.ruby-lang.org/ ?